src/hssm/distribution_utils/dist.py

```diff
@@ -160,6 +163,11 @@ def rng_fn(
                 size = args[-1]
                 args = args[:-1]
 
+            # Although we got around the ndims_supp issue, the size parameter passed
+            # here is still an array with one element. We need to take it out.
+            if not np.isscalar(size):
+                size = np.squeeze(size)
```

Comment:
Reading this line I went down a bit and also looked at line 185 ++.
Do we report somewhere the complete logic on how we deal with sizes? 
Just for future reference, I think we should record this as comments.

Reply:
Added this as Note in the docstring